source/common/tcp_proxy/upstream.h

```diff
@@ -208,6 +208,7 @@ class HttpUpstream : public GenericUpstream, protected Http::StreamCallbacks {
   };
   DecoderShim response_decoder_;
   Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks_;
+  const Http::CodecType type_;
```

Comment:
Is there a specific architectural reason for this change? Previously different behavior was encapsulated in different classes and now it is encapsulated in the same class using state for differentiation.

This change is arguably an antipattern.

Reply:
the long term plan is to move all of this code to using an upstreamRequest class which as far as I remember was easier with the combined class.   I the prototype is linked somewhere in the issue if you take a look